x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -377,6 +380,16 @@ export class MapEmbeddable
     this._isSharable = isSharable;
   }
 
+  public onReset(lastInput: MapByValueInput) {
+    // TODO add some test / typings that ensure that every piece of Map Input which can be changed in _handleStoreChanges is also reset properly in this function
+    const { mapCenter, isLayerTOCOpen, openTOCDetails, hiddenLayers } = lastInput;
+    if (mapCenter) this._savedMap.getStore().dispatch(setGotoWithCenter(mapCenter));
+    if (isLayerTOCOpen !== undefined)
+      this._savedMap.getStore().dispatch(setIsLayerTOCOpen(isLayerTOCOpen));
+    if (hiddenLayers) this._savedMap.getStore().dispatch<any>(setHiddenLayers(hiddenLayers));
+    if (openTOCDetails) this._savedMap.getStore().dispatch(setOpenTOCDetails(openTOCDetails));
```

Comment:
This does not handle the case where a by-value map is edited in the maps app. For example, we may change the color in a layer. If we switch to view mode and discard changes, the unsaved edit does not get restored. Would it make more sense to destroy the map embeddable and recreate it? @nreese 

I wonder how other embeddables handle this?

Reply:
They usually handle it in a spotty way, but I think the main thing is that the embeddable input _is_ the source of truth for most embeddables.